### PR TITLE
[WasmFS][NFC] Simplify the async JS backend code

### PR DESCRIPTION
**Pull request by tlively**
Remove the CallbackState argument in favor of directly passing outparam pointers
for results and remove the callback function in favor of passing the
em_proxying_ctx and directly calling emscripten_proxy_finish on it. Also remove
the unnecessary `runtimeKeepalivePush` and `runtimeKeepalivePop` calls since the
`ProxyWorker` utility already takes care of keeping the runtime alive.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #16892** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16892" target="_blank">Graphite</a> 
    * **PR #16893** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16893" target="_blank">Graphite</a> 
      * **PR #16894** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16894" target="_blank">Graphite</a>  (this PR)
        * **PR #16895** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16895" target="_blank">Graphite</a> 
          * **PR #16896** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16896" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/emscripten-core/emscripten/16894?utm_source=stack-comment).

**Review by kripken — APPROVED** [review]
lgtm % comment

File: tools/deps_info.py [hunk 1]
```diff
@@ -221,5 +221,6 @@ def get_deps_info():
 
   if settings.WASMFS:
     _deps_info['wasmfs_create_opfs_backend'] = ['emscripten_proxy_finish']
+    _deps_info['wasmfs_create_fetch_backend'] = ['emscripten_proxy_finish']
```
[the remark this answers is not in the document]
Comment (tlively) [reply tools/deps_info.py:225]: Sounds good. I'll unconditionally export it in that case from emcc.py.

**Comment by tlively** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/emscripten-core/emscripten/16894?utm_source=stack-comment).


Job Reference: iF1KqnaMxvB9QQZegkQ0

**Pull request merged**